### PR TITLE
Fix bug in lazy CutSet subset with last

### PR DESCRIPTION
**Pull request by desh2608**
This was raised in https://github.com/lhotse-speech/lhotse/discussions/1018#discussioncomment-5552948.

### CODE REVIEW

File: lhotse/cut/set.py [hunk 1]
```diff
@@ -996,8 +996,7 @@ def subset(
                     f"CutSet has only {len(self)} items but last {last} required; not doing anything."
                 )
                 return self
-            cut_ids = list(self.ids)[-last:]
-            return CutSet.from_cuts(self[cid] for cid in cut_ids)
+            return CutSet.from_cuts(islice(self, len(self) - last, len(self)))
```
Comment (pzelasko) [inline lhotse/cut/set.py:999]: Can you assign len(self) to a variable to avoid double iteration for lazy cut sets?

Comment (desh2608) [reply lhotse/cut/set.py:999]: Yeah, good point.

Comment (desh2608) [reply lhotse/cut/set.py]: Done.

**Review by pzelasko — APPROVED** [review]

**Comment by pzelasko** [comment]
Thanks!

**Pull request merged**